packages/cli/src/workflows/workflow.service.ee.ts

```diff
@@ -46,9 +50,17 @@ export class EnterpriseWorkflowService {
 	) {
 		const em = entityManager ?? this.sharedWorkflowRepository.manager;
 
-		const projects = await em.find(Project, {
+		let projects = await em.find(Project, {
 			where: { id: In(shareWithIds), type: 'personal' },
+			relations: { sharedWorkflows: true },
 		});
+		// filter out all projects that already own the workflow
+		projects = projects.filter(
+			(p) =>
+				!p.sharedWorkflows.some(
+					(swf) => swf.workflowId === workflow.id && swf.role === 'workflow:owner',
+				),
+		);
```

Comment:
This is not covered by a test.

Reply:
@despairblue following the pattern from credential testing I made this
https://github.com/n8n-io/n8n/pull/12524/commits/e5aeee86dd279d975a715c1b30796bca4e8a6218